src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionInfo.cs

```diff
@@ -99,6 +99,15 @@ public ICollection<string> AllConnectionTypes
             }
         }
 
+        public bool HasConnectionType(string connectionType)
+        {
+            if (connectionType == null)
+            {
+                connectionType = ConnectionType.Default;
+            }
+            return ConnectionTypeToConnectionMap.ContainsKey(connectionType);

```

Comment:
Yep. We can reassess in the future. This on is a concurrent dictionary, but the entire logic is in a different class (where it should be). We need to take a bigger look at the whole thing. For instance, we'd probably need to add a method here that lets you operate on the map by passing in a function. Then in the calling class refactor to use this approach. For this PR that's too much. But... we need to fix in a separate PR and actually improve the design in the ConnectionService.